src/OrchardCore.Modules/OrchardCore.Users/Handlers/ScriptExternalLoginEventHandler.cs

```diff
@@ -54,6 +61,32 @@ public async Task UpdateRoles(UpdateRolesContext context)
                 dynamic evaluationResult = _scriptingManager.Evaluate(script, null, null, null);
                 context.RolesToAdd.AddRange((evaluationResult.rolesToAdd as object[]).Select(i => i.ToString()));
                 context.RolesToRemove.AddRange((evaluationResult.rolesToRemove as object[]).Select(i => i.ToString()));
+
+                if (evaluationResult.claimsToUpdate is not null)
+                {
+                    var claimsToUpdate = ((JsonObject)JObject.FromObject(evaluationResult.claimsToUpdate)).Deserialize<List<UserClaim>>();
```

Comment:
For me this will always fail if ClaimsToUpdate is not set by the script. The null check above will always pass and this will throw a `System.InvalidOperationException`: The element must be of type 'Object'.